src/inmanta/util.py

```diff
@@ -750,6 +750,52 @@ def ensure_event_loop() -> asyncio.AbstractEventLoop:
         return new_loop
 
 
+class OnAccess(enum.EnumMeta):
```

Comment:
Both this class and `DeprecatedEnum` are still raising a lot of mypy errors, I could use some help with dealing with those

Reply:
I can have a look but let's first reach a conclusion on this vs a pydantic validator and then come back to this.

Reply:
Removed them both